rclpy/src/rclpy/_rclpy.c

```diff
@@ -1646,6 +1646,53 @@ rclpy_publish(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Publish a serialized message
+/**
+ * Raises ValueError if pypublisher is not a publisher capsule
+ * Raises RuntimeError if the message cannot be published
+ *
+ * \param[in] pypublisher Capsule pointing to the publisher
+ * \param[in] pymsg serialized message to send
+ * \return NULL
+ */
+static PyObject *
+rclpy_publish_raw(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pypublisher;
+  PyBytesObject * pymsg;
+  const char * pymsg_buffer;
+  Py_ssize_t pymsg_size;
+
+  if (!PyArg_ParseTuple(args, "OS", &pypublisher, &pymsg)) {
+    return NULL;
+  }
+
+  PyBytes_AsStringAndSize((PyObject *)pymsg, &pymsg_buffer, &pymsg_size);
```

Comment:
When I compile I see a warning at this line:

```
warning: passing argument 2 of ‘PyBytes_AsStringAndSize’ from incompatible pointer type [-Wincompatible-pointer-types]
   PyBytes_AsStringAndSize((PyObject *)pymsg, &pymsg_buffer, &pymsg_size);
                                              ^
...

/usr/include/python3.6m/bytesobject.h:103:17: note: expected ‘char **’ but argument is of type ‘const char **’
 PyAPI_FUNC(int) PyBytes_AsStringAndSize(
```

I think `pymsg_buffer` should be declared as `char *` above (no `const`).

Reply:
I've removed the identifier, also had to update the function to use `rclpy_handle_get_pointer_from_capsule` to get the publisher

Reply:
Thanks!